src/Workspaces/Core/Portable/Log/RoslynEventSource.LogBlock.cs

```diff
@@ -0,0 +1,162 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.Internal.Log
+{
+    internal partial class RoslynEventSource
+    {
+        /// <summary>
+        /// Logs an informational block with given <paramref name="entity"/>'s <see cref="object.ToString"/> representation as the message
```

Comment:
Is the intent of the "object entity" that we avoid realizing the string if the user doesn't have telemetry on?

Reply:
Yes, especially given this telemetry is only enabled when user explicitly enables RoslynEventSource by following the steps at https://aka.ms/reportPerf